### PR TITLE
Pin python version and environment

### PR DESCRIPTION
**Pull request by huppd**
- fix python version to commonly use the same
- use pinned environment for CI
- improve README.md
- hotfix failing pylint
- update gitignore

### CODE REVIEW

File: .github/workflows/pre-commit.yml [hunk 1]
```diff
@@ -19,16 +19,16 @@ jobs:
     - name: Set up Python
       uses: actions/setup-python@v2
       with:
-        python-version: 3.11.0
+        python-version: 3.10.8
```
Comment (stelliom) [inline .github/workflows/pre-commit.yml:22]: Why exactly 3.10.8 instead of the more recent 3.11?

File: setup_env.sh [hunk 2]
```diff
@@ -13,7 +13,7 @@ DEFAULT_ENV_NAME="probtest"
 
 # Default options
 ENV_NAME="${DEFAULT_ENV_NAME}"
-PYVERSION=3.11
+PYVERSION=3.10
```
Comment (stelliom) [inline setup_env.sh:16]: Why not 3.10.8?

File: .github/workflows/pre-commit.yml (hunk 1 above)
Comment (huppd) [reply .github/workflows/pre-commit.yml:22]: 3.10.8 is the version that is the most recent version that we get on balfrin in the user environment, which is used also by buildbot. So we are testing the same version that we use on balfrin. Once we made probtest independent from that we can use more recent version.

File: setup_env.sh (hunk 2 above)
Comment (huppd) [reply setup_env.sh]: Good point, even better.

File: README.md [hunk 3]
```diff
@@ -107,7 +107,8 @@ setup scripts:
 
 ```console
 ./setup_miniconda.sh
-./setup_env.sh -n probtest -u
+source miniconda/bin/activate
```
Comment (stelliom) [inline README.md:110]: I think this is better, but we could also just do `./setup_miniconda.sh -u`, right?

File: README.md [hunk 4]
```diff
@@ -107,7 +107,8 @@ setup scripts:
...
+./setup_env.sh
```
Comment (stelliom) [inline README.md:111]: Maybe we could add a note saying something like: "For an unpinned environment setup use the `-u` option".

File: .github/workflows/pre-commit.yml (hunk 1 above)
Comment (stelliom) [reply .github/workflows/pre-commit.yml:22]: I see, ok makes sense :+1:

File: README.md (hunk 3 above)
Comment (huppd) [reply README.md:110]: Yes, you are right that would also work. This is just my prefered way of doing it :)
But we can mention both so everyone can choose for themself.

File: README.md (hunk 4 above)
Comment (huppd) [reply README.md]: I agree, I moved it into its own section, and make it a bit more verbose.

Comment (stelliom) [reply README.md]: Perfect, thank you!

**Review by stelliom — APPROVED** [review]
This looks good to me after the last changes :+1:

**Pull request merged**